### PR TITLE
Adds Recipe to reverse engineer White Wool into String

### PR DESCRIPTION
**Pull request by pmardle**
String is a RARE commodity as we need a lot for guards/fishermen.  This recipe lets the player convert 1 white wool block into 4 string (shapeless)

Requested by the community

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/util/RecipeHandler.java [hunk 1]
```diff
@@ -10,6 +10,7 @@
 import net.minecraft.item.ItemStack;
 import net.minecraftforge.fml.common.registry.GameRegistry;
 import net.minecraftforge.oredict.ShapedOreRecipe;
+import net.minecraftforge.oredict.ShapelessOreRecipe;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/util/RecipeHandler.java]: MINOR Remove this unused import 'net.minecraftforge.oredict.ShapelessOreRecipe'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 30 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 28 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RecipeHandler.java#L100](https://github.com/Minecolonies/minecolonies/blob/ff52dc0ce1ae8845ba96b27cc3e47cfa39d2ef9c/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L100): Remove this unused private "enableInDevelopmentFeatures" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L119](https://github.com/Minecolonies/minecolonies/blob/ff52dc0ce1ae8845ba96b27cc3e47cfa39d2ef9c/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L119): Remove this unused private "addHutRecipe" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/ff52dc0ce1ae8845ba96b27cc3e47cfa39d2ef9c/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/ff52dc0ce1ae8845ba96b27cc3e47cfa39d2ef9c/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/ff52dc0ce1ae8845ba96b27cc3e47cfa39d2ef9c/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/ff52dc0ce1ae8845ba96b27cc3e47cfa39d2ef9c/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/ff52dc0ce1ae8845ba96b27cc3e47cfa39d2ef9c/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/ff52dc0ce1ae8845ba96b27cc3e47cfa39d2ef9c/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/ff52dc0ce1ae8845ba96b27cc3e47cfa39d2ef9c/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/ff52dc0ce1ae8845ba96b27cc3e47cfa39d2ef9c/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 29 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 27 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RecipeHandler.java#L99](https://github.com/Minecolonies/minecolonies/blob/0cca79aea3d763a38088ab652bb6bca5d92bdaaf/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L99): Remove this unused private "enableInDevelopmentFeatures" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L118](https://github.com/Minecolonies/minecolonies/blob/0cca79aea3d763a38088ab652bb6bca5d92bdaaf/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L118): Remove this unused private "addHutRecipe" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/0cca79aea3d763a38088ab652bb6bca5d92bdaaf/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/0cca79aea3d763a38088ab652bb6bca5d92bdaaf/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/0cca79aea3d763a38088ab652bb6bca5d92bdaaf/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/0cca79aea3d763a38088ab652bb6bca5d92bdaaf/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/0cca79aea3d763a38088ab652bb6bca5d92bdaaf/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/0cca79aea3d763a38088ab652bb6bca5d92bdaaf/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/0cca79aea3d763a38088ab652bb6bca5d92bdaaf/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/0cca79aea3d763a38088ab652bb6bca5d92bdaaf/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BrianEdmonds** [comment]
It's pretty common for crusher type machines in tech mods to offer this
conversion, at this rate.  I'd suggest a less productive rate for a
simple crafting recipe, perhaps two string for one wool?

**Comment by MineShadow7** [comment]
Waiting for pmardle to response

**Comment by BrianEdmonds** [comment]
Thinking about it further, this would make an interesting colonist, to
add a spinner, who could spin wool into string, at the same conversion
rate as the machines, four to one.  It would fit nicely into a low tech
pipeline with a shepherd to look after some sheep, and a bowyer to turn
wood from the lumberjack, and string from the spinner into bows for the
rangers.

**Comment by marchermans** [comment]
I don't think this should be done this way. Either introduce a way to get it from one of our colonists, or just build a mob farm (plenty of tutorials out there that show how to build one in vanilla).

**Review by marchermans — COMMENTED** [review]
Find a different solution for this. In my opinion this id not even an issue. We should focus on other things.

**Review by marchermans — CHANGES_REQUESTED** [review]
Damnit stupid phone did not mark it as requesting changes.....

**Comment by MineShadow7** [comment]
So what should be changed?

**Comment by Raycoms** [comment]
I think as soon as we add a sheepherder he could get randomly string and wool from the sheeps?

**Comment by pmardle** [comment]
The issue which several people have raised with me is for guards, most
people use rangers a lot and the string requirements for their bows are
significant as they tend to get through them quickly.  Currently the only
way to get string (and forget about other mods, because our mod has to
function well stand alone - not everyone plays in modpacks) - is from
cobwebs in mineshafts and from spiders..

This was only adding a simple recipe in to reverse engineer the 4 string =
1 wool recipe.  Happy for it to be 2 string per wool if you like.

Adding a whole worker with all the requirements to build a house and feed
them is overkill just to get string.  more than happy for it to be a by
product of the shepherd in the future though, but for now I think a simple
recipe like this fixes a hole for now and we could remove the recipe later
on when the Shepard can do it.

**Comment by zekermarco** [comment]
I agree with pmardle. Wool is easier to get through sheep, so the players won't be having a hard time searching for string.

**Comment by Raycoms** [comment]
:shipit: for me

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1292/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

File: src/main/java/com/minecolonies/coremod/util/RecipeHandler.java [hunk 2]
```diff
@@ -31,9 +31,11 @@
      * The amount of items returned by certain crafting recipes
      */
     private static final int ONE_FORTH_OF_A_STACK = 16;
+    private static final int FOUR = 4;
```
Comment (marvin-bitterlich) [inline src/main/java/com/minecolonies/coremod/util/RecipeHandler.java]: rename this to declare intend not value please

**Review by marvin-bitterlich — CHANGES_REQUESTED** [review]

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 29 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 27 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [RecipeHandler.java#L99](https://github.com/Minecolonies/minecolonies/blob/9afbda1fb162c63d56a0128eab42cbb7dc9fe064/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L99): Remove this unused private "enableInDevelopmentFeatures" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [RecipeHandler.java#L118](https://github.com/Minecolonies/minecolonies/blob/9afbda1fb162c63d56a0128eab42cbb7dc9fe064/src/main/java/com/minecolonies/coremod/util/RecipeHandler.java#L118): Remove this unused private "addHutRecipe" method. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/9afbda1fb162c63d56a0128eab42cbb7dc9fe064/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/9afbda1fb162c63d56a0128eab42cbb7dc9fe064/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/9afbda1fb162c63d56a0128eab42cbb7dc9fe064/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/9afbda1fb162c63d56a0128eab42cbb7dc9fe064/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/9afbda1fb162c63d56a0128eab42cbb7dc9fe064/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/9afbda1fb162c63d56a0128eab42cbb7dc9fe064/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/9afbda1fb162c63d56a0128eab42cbb7dc9fe064/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/9afbda1fb162c63d56a0128eab42cbb7dc9fe064/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by ghost** [comment]
Ideally guards should be able to get their own string by killing spiders. Maybe if their ai changed a bit they would get better at it.

**Pull request closed without merging**